### PR TITLE
Enhancement features development

### PR DESCRIPTION
**Pull request by jinningwang**
- Add function signature ``attr='v'`` to method ``Model.alter`` for altering parameter values without manual
  per unit conversion
- Add following methods to `GroupBase` for easier usage: ``get_all_idxes``, ``alter``, ``as_dict``, and ``as_df``
- Add two demo "add_RenGen.ipynb" and "replace_SynGen.ipynb" to show how to do renewable penetration via code.
  See folder ``andes/examples/demonstration``
- Add a demo "manipulate_params.ipynb" to compare the differnce between ``set`` and ``alter``
- Extend ``SynGen`` common parameters with ``bus``, ``gen``, ``Sn``, ``Vn``, and ``fn``
- Extend ``StaticGen`` common parameters with ``bus``

### CODE REVIEW

**Comment by jinningwang** [comment]
This PR, if successful, can address https://github.com/CURENT/andes/issues/579

**Comment by jinningwang** [comment]
@cuihantao This PR is ready to merge if it looks good to you

**Comment by cuihantao** [comment]
Jinning, it looks great! Let me take a closer look later today. I'll merge it very soon.

File: andes/core/model/model.py [hunk 1]
```diff
@@ -546,17 +546,43 @@ def alter(self, src, idx, value):
...
+        >>> ss.GENCLS.get(src='M', idx=2, attr='v')
+        1.0
         """
 
         instance = self.__dict__[src]
 
-        if hasattr(instance, 'vin') and (instance.vin is not None):
-            self.set(src, idx, 'vin', value)
+        if not hasattr(instance, 'vin'):
+            raise AttributeError(f"{self.class_name}.{src} has no `vin` attribute.")
```
Comment (jinningwang) [inline andes/core/model/model.py]: Maybe I can make a small change there. If there is no `vin`, fall back to `v` rather than raising an error.

**Comment by cuihantao** [comment]
Thanks, Jinning!

**Pull request merged**